faker/providers/internet/__init__.py

```diff
@@ -178,6 +178,22 @@ def ipv4(self, network=False):
             address = str(ip_network(address, strict=False))
         return address
 
+    def ipv4_private(self, klass=None, network=False):
```

Comment:
Could we merge this logic into the `ipv4` method and have non-private address classes? I'm thinking something like:

```python
def ipv4(self, network=False, private=None, address_class=None):
    if private is None and address_class is None:
        # pretty much the old logic
    if private is True:
        classes = {
            'a': (167772160, 184549375, 24),
            'b': (2886729728, 2887778303, 20),
            'c': (3232235520, 3232301055, 16),
        }
  if private is False:
        classes = {
            'b': (2886729728, 2887778303, 16),
            'c': (3232235520, 3232301055, 8),
        }
```

Reply:
should we have the same for IPV6 addresses?

Reply:
> I would use address_class instead of klass
Ok

> Could we merge this logic into the ipv4 method and have non-private address classes?
I don't think we should merge Unaware/Public/Private. 

> should we have the same for IPV6 addresses?
Yes of course, but another PR ?

Reply:
> I don't think we should merge Unaware/Public/Private.

Could you elaborate why that would be a bad idea? I'd like to keep the number of methods to a minimum

Reply:
Yes, of course,
1. To get private or public is a little complex and separated code is better.
2. Like the `person` provider with male & female, we should provide simple way to get private or public IP.